src/ui/common/src/components/tables/PaginatedSearchTable.tsx

```diff
@@ -125,53 +153,196 @@ export const PaginatedSearchTable: React.FC<PaginatedSearchTableProps> = ({
-                  />
-                </Box>
-              ),
-              endAdornment: (
-                <Box
-                  marginLeft="8px"
-                  color={theme.palette.gray[600]}
-                  sx={{
-                    '&:hover': {
-                      cursor: 'pointer',
-                      color: theme.palette.black,
-                    },
-                  }}
-                  onClick={() => {
-                    setSearchQuery('');
-                  }}
+        <Box mb="8px">
+          <Box marginBottom="8px" display="flex">
+            <TextField
+              placeholder="Search"
+              value={searchQuery}
+              onChange={(event) => setSearchQuery(event.target.value)}
+              id="outlined-basic"
+              variant="outlined"
+              size="small"
+              sx={{ minWidth: '300px' }}
+              InputProps={{
+                startAdornment:
+                  searchQuery === '' ? (
+                    <Box marginRight="8px">
+                      <FontAwesomeIcon
+                        icon={faSearch}
+                        color={theme.palette.gray[600]}
+                      />
+                    </Box>
+                  ) : (
+                    <Box
+                      marginRight="8px"
+                      color={theme.palette.gray[600]}
+                      sx={{
+                        '&:hover': {
+                          cursor: 'pointer',
+                          color: theme.palette.black,
+                        },
+                      }}
+                      onClick={() => {
+                        setSearchQuery('');
+                      }}
+                    >
+                      <FontAwesomeIcon icon={faX} />
+                    </Box>
+                  ),
+              }}
+            />
+
+            {sortColumns && sortColumns.length > 0 && (
+              <Button
+                size="small"
+                color="secondary"
+                endIcon={<FontAwesomeIcon icon={faArrowUpShortWide} />}
+                sx={{ ml: 1 }}
+                onClick={(e) => setSortMenuAnchor(e.currentTarget)}
+              >
+                Sort
+              </Button>
+            )}
+
+            <Menu
+              open={!!sortMenuAnchor}
+              anchorEl={sortMenuAnchor}
+              onClose={() => setSortMenuAnchor(null)}
+            >
+              {sortColumns.map((column, idx) => (
+                // Note that the menu doesn't seem to like non-numerical values, so we use the idx here.
+                <MenuItem
+                  key={column.name}
+                  value={idx}
+                  onClick={(e) => setSortTypeMenuAnchor(e.currentTarget)}
                 >
-                  <FontAwesomeIcon icon={faX} />
+                  {column.name}
+                </MenuItem>
+              ))}
+            </Menu>
+
+            <Menu
```

Comment:
This PaginatedSearchTable is starting to get a little long with the search / menu functionality added in. 

Doesn't have to be done right now, but can you add a small refactoring task to turn them into their own respective components?